apps/shipments/views.py

```diff
@@ -206,3 +206,18 @@ def create(self, request, *args, **kwargs):
         permission_link = PermissionLink.objects.create(**serializer.validated_data)
 
         return Response(PermissionLinkSerializer(permission_link).data, status=status.HTTP_201_CREATED)
+
+
+class ShipmentHistoryListView(viewsets.ViewSet):
+    permission_classes = ((IsOwnerOrShared,) if settings.PROFILES_ENABLED
+                          else (permissions.AllowAny,))
+    http_method_names = ['get', ]
+
+    def list(self, request, *args, **kwargs):
+
+        shipment = Shipment.objects.get(id=kwargs['shipment_pk'])
+
+        queryset = shipment.history.all()
```

Comment:
I think we are going to have to come up with a pagination solution as this is going to get huge quickly. But we can push that to the backlog for now.